### DISCUSSION TITLE
Bring back Tiles to pin on Start Menu

### DISCUSSION THREAD

**Discussion by RenanMDP** [valinet/ExplorerPatcher#60] [Ideas]
Hello! Thanks for your awesome work on this feature!

Any chance of bringing back Tiles to pin on the start menu? I actually use them to keep my desktop clean and provide easy access to all my apps.

Windows 11 made them worse by only allowing a few per page and no more grouping. 😢

**Comment by valinet** [comment]
Hi

I will quote my reply from an email conversation I had with someone else on the same topic. There was an issue opened around here as well talking about this, but I cannot seem to find it. He asked about reviving the Windows 10 Start menu, as the code is mostly still in the system.

> Indeed, I looked on it, the code is still there. Well, most of it is intact. Most of it. The thing is, it does not work properly. In the brief time that I run and looked at it, there are at least 2/3 issues:
>
> * Right clicking any item crashes the Start menu
> * There is a delay of 1-2 seconds when the desktop does not update after the Start menu is dismissed (I know, weird)
> * Live tiles seem kind of broken - they act weird, changing often but not really updating the information displayed
>
> Of course, I looked on the disassembly as well; the check for the registry entry has been replaced with basically a call to a function called "IsPPIEnabled" (I don't know what "PPI" stands for in this context). That function boils down to checking whether the edition reported by "GetProductInfo" is 119 (or 0x77). If it is, it returns 1, otherwise 0. That number is nowhere to be found in the documentation, so I don't know what Windows version it refers to, but maybe I missed something. I tested and confirmed that, indeed, if you make "IsPPIEnabled" return 1 (or patch the IAT to have "GetProductInfo" provide a Windows version with number 119), the Windows 10 Start menu shows in all its glory, but with the issues mentioned above. Now, there are two ways to achieve this:
>
> * The way I did it, was to modify the return of "IsPPIEnabled" (modify the "rax" register after the call) using a debugger (I used WinDbg)
> * Alternatively, use a "StartMenuExperienceHost.exe" from an older build (like .51, for example). You can also try with a "StartDocked.dll" from that build as well ("StartDocked.dll" is what hosts the logic for the menu, apparently)
>
> The problems occur regardless if using files from the latest build or from .51, so clearly something else this depends on changed. I don't know. Maybe you can investigate this, starting from this pointers that I gave you.
>
> Also, regarding enabling this automatically using ExplorerPatcher, a funny thing happened: that check is so early in the execution, that by the time "dxgi.dll" gets injected, the program has long ago decided to use the new menu, so my in-memory patching is useless. "dxgi.dll" works better for File Explorer as that calls very early one of the DXGI functions. So yeah, even hooking this is pretty hard. Hooking may have to be done substituting another DLL, avoiding KnownDLLs (list is in "HKML\SYSTEM\CurrentControlSet\Control\Session Manager\KnownDLLs") etc.
>
> So yeah, that's the story so far. And that's the story in general with Windows 11. Not only do they remove features, but they also break them just a bit to make them not work well if enabled back somehow. For example, for Win+X in ExplorerPatcher, I leveraged a lot of code from their DLLs - they just removed the code in the actual method that showed the menu, so I had to port that over from an older version of the DLL ("twinui.pcshell.dll").
>
> I will maybe look more into it some day if I have time, but I can't make any promises; I don't know. With a few fixes, the new menu is fine as well, matches the aesthetic with search, can be centered and so on. They should just add officially some common sense options, like the ones from my patcher, plus things like resizing the menu, pining a few icons in the app list etc., then it is all fine by me.

In the mean time, some people pointed out that `IsPPIEnabled`, specifically PPI, stands for Windows Team, an edition designed for the Surface Hub.

So yeah, that's the story so far. I am not against implementing this in any way, it just takes time and it is not at the top of my priority list, but of course, if anyone wants to take the lead, I am more than happy. If not, hopefully at some point this could be implemented as well.

**Comment by marrahouser** [comment]
First: Thanks for your work. It is awesome tool which returned me to Win11 after quick dissapointment and downgrade after few hours.

And i am also hooking here. Apps and their tiles (small squares without names) lined in groups by type (online tools, graphic tools, audio tools, office apps, games...) and precisely set background border around them is *chef kiss* for my Windows 10 experience with very fast click-click opening and clean desktop. I really don't understand why Microsoft ruins it and i hope that they will change their minds or you will find the solution.

![image](https://user-images.githubusercontent.com/34096244/143662287-a5455380-ffae-4706-8511-b84dbfe75582.png)

**Comment by sunrastardust** [comment]
High, thank you so much for your work! Do you think you could add an option to remove the recommended section from the start menu, to be able to use the height for the pinned apps? I don´t get why they removed the option to resize the start menu, but if I could do what I´ve described, I could at least add all my apps here which I use regularly on my productive system. 

![image](https://user-images.githubusercontent.com/1408066/144704815-10b34af1-a73b-4d77-ae7e-c93f07b66ccb.png)

**Reply by valinet** [reply]
I have looked on it some time ago, I did not find anything relevant that could be changed. I don't have much experience with this UI framework, maybe it could be somehow done by walking and altering the UI tree; unfortunately, I don't know much about it. As for patching the disassembly, sure, but making sense of especially all the new UI is like taking a trip to North Korea, it's that closed and obfuscated... yet Microsoft doesn't miss a single opportunity to brag how open they are. Anyway, should I or anyone else find anything more advanced regarding this, I will take a look, of course.

**Comment by sunrastardust** [comment]
Alright, thanks for your answer! I was looking into the sources, but as a simple C# an web dev, I have no glue what´s going on there. :-D

**Reply by valinet** [reply]
You probably mean the disassembly, it would have been a million times better if we had access to the sources…

**Comment by Wolf-Shadow777** [comment]
Hi, this is not a problem so much as a request . I am running windows 10, and have the taskbar centered. When first started or if explorer.exe is restarted, the start menu starts on the left side and then quickly gets centered . After which, it starts at centered after that.
Is it something that could be looked into? As I said, not so much a problem but would be nice if it worked like windows 11.

**Reply by Poopooracoocoo** [reply]
It sounds unrelated to this feature request. You should make your own post.
